packages/react-components/react-calendar-compat/src/components/Calendar/Calendar.types.ts

```diff
@@ -0,0 +1,251 @@
+   * Defines when the first week of the year should start.
+   * @default FirstWeekOfYear.FirstDay
+   */
+  firstWeekOfYear?: FirstWeekOfYear;
+
+  /**
+   * Whether the month picker is shown beside the day picker or hidden.
+   * @default true
+   */
+  isMonthPickerVisible?: boolean;
+
+  /**
+   * Whether the day picker is shown beside the month picker or hidden.
+   * @default true
+   */
+  isDayPickerVisible?: boolean;
+
+  /**
+   * Show month picker on top of date picker when visible.
+   * @default false
+   */
+  showMonthPickerAsOverlay?: boolean;
+
+  /**
+   * Whether the "Go to today" link should be shown or not
+   */
+  showGoToToday?: boolean;
+
+  /**
+   * Whether the calendar should show the week number (weeks 1 to 53) before each week row
+   * @default false
+   */
+  showWeekNumbers?: boolean;
+
+  /**
+   * Localized strings to use in the Calendar
+   */
+  strings?: CalendarStrings;
+
+  /**
+   * Apply additional formatting to dates, for example localized date formatting.
+   */
+  dateTimeFormatter?: DateFormatting;
+
+  /**
+   * If set the Calendar will not allow navigation to or selection of a date earlier than this value.
+   */
+  minDate?: Date;
+
+  /**
+   * If set the Calendar will not allow navigation to or selection of a date later than this value.
+   */
+  maxDate?: Date;
+
+  /**
+   * If set the Calendar will not allow selection of dates in this array.
+   */
+  restrictedDates?: Date[];
+
+  /**
+   * Whether the calendar should show 6 weeks by default.
+   * @default false
+   */
+  showSixWeeksByDefault?: boolean;
+
+  /**
+   * The days that are selectable when `dateRangeType` is `WorkWeek`.
+   * If `dateRangeType` is not `WorkWeek` this property does nothing.
+   * @default [Monday,Tuesday,Wednesday,Thursday,Friday]
+   */
+  workWeekDays?: DayOfWeek[];
+
+  /**
+   * Whether the month picker should highlight the current month
+   * @default false
+   */
+  highlightCurrentMonth?: boolean;
+
+  /**
+   * Whether the month picker should highlight the selected month
+   * @default false
+   */
+  highlightSelectedMonth?: boolean;
+
+  /**
+   * Whether the close button should be shown or not
+   * @default false
+   */
+  showCloseButton?: boolean;
+
+  /**
+   * Allows all dates and buttons to be focused, including disabled ones
+   * @default false
+   */
+  allFocusable?: boolean;
+}
+
+/**
+ * @internal
+ */
+export interface CalendarStyleProps {
+  /**
+   * Custom CSS class for the calendar.
+   */
+  className?: string;
+
+  /**
+   * Whether the month picker is visible
+   */
+  isMonthPickerVisible?: boolean;
+
+  /**
+   * Whether the day picker is visible
+   */
+  isDayPickerVisible?: boolean;
+
+  /**
+   * Whether only month picker is shown
+   */
+  monthPickerOnly?: boolean;
+
+  /**
+   * Whether the month picker is overlaid on the day picker
+   */
+  showMonthPickerAsOverlay?: boolean;
+
+  /**
+   * @deprecated Use `overlaidWithButton`
+   */
+  overlayedWithButton?: boolean;
```

Comment:
For now I'll leave as is since I don't want to bloat the PR, but I'll make a pass on the props to see if it's still relevant. Though we might keep it for backwards compat. Thanks for pointing it out :)